packages/core/stories/dropdown/dropdown.stories.tsx

```diff
@@ -341,3 +341,70 @@ export const ObjectValue: StoryFn<DropdownProps<Person>> = () => {
     </Dropdown>
   );
 };
+export const SelectAll: StoryFn<DropdownProps> = (args) => {
+  const [selected, setSelected] = useState<string[]>([]);
+  const allSelectedOptionValue = "all";
+
+  const handleSelectionChange: DropdownProps["onSelectionChange"] = (
+    event,
+    newSelected
+  ) => {
+    let newOptionsSelected = [...newSelected];
+    const wasAllSelected = selected.includes(allSelectedOptionValue);
+    const isAllSelected = newOptionsSelected.includes(allSelectedOptionValue);
+
+    if (wasAllSelected) {
+      if (isAllSelected) {
+        newOptionsSelected = newOptionsSelected.filter(
+          (el) => el !== allSelectedOptionValue
+        );
+      } else {
+        newOptionsSelected = [];
+      }
+    } else if (
+      isAllSelected ||
+      (!isAllSelected && newOptionsSelected.length === usStates.length)
+    ) {
+      newOptionsSelected = [...usStates, allSelectedOptionValue];
```

Comment:
I think order of `allSelectedOptionValue` is the reason of losing focus on selection. `allSelectedOptionValue` in front resolves it